docs/cookbook/languages/python.md

```diff
@@ -0,0 +1,289 @@
+---
+title: Python
+description: Common questions and solutions for using Python with Flox
+---
+
+# Python
+
+This provides an overview of how to create Flox environments for new or existing Python projects.
+
+## Using Flox in a new project
+
+Getting started with Flox is super simple. First, create and/or `cd` into your project directory.
+
+```
+mkdir new-python-project && cd new-python-project
+```
+
+Next, initialize your Flox environment:
+
+```
+flox init
+```
+
+This will suggest a few commands you can run next:
+
+```
+$ flox init
+✨ Created environment 'new-python-project' (x86_64-linux)
+
+Next:
+  $ flox search <package>    <- Search for a package
+  $ flox install <package>   <- Install a package into an environment
+  $ flox activate            <- Enter the environment
+  $ flox edit                <- Add environment variables and shell hooks
+```
+
+This Flox environment is now ready to be populated with packages.
+
+### Select a Python interpreter
+
+To begin, we need a Python interpreter. For this example, we will be using Python 3.11. Search for the version of Python your project requires, omitting the dot between major and minor numbers:
+
+```
+flox search python311
+```
+
+This will show you the list of packages matching the major and minor version of Python you specified:
+
+```
+$ flox search python311
+python311      High-level dynamically-typed programming language
+python311Full  High-level dynamically-typed programming language
+
+Use 'flox show <package>' to see available versions
+```
+
+To see the specific versions of Python available in the Flox Catalog, use the `flox show` command:
+
+```
+flox show python311Full
+```
+
+This will list all of the releases that match the major and minor numbers you provided:
+
+```
+$ flox show python311Full
+python311Full - High-level dynamically-typed programming language
+    python311Full@python3-3.11.9
+    python311Full@python3-3.11.8
+    python311Full@python3-3.11.7
+    python311Full@python3-3.11.6
+    python311Full@python3-3.11.5
+    python311Full@python3-3.11.4
+    python311Full@python3-3.11.3
+    python311Full@python3-3.11.2
+    python311Full@python3-3.11.1
+```
+
+At this point, you can install the latest version by running `flox install` without a version specified:
+
+```
+flox install python311Full
+```
+
+This will show a message indicating the package was successfully installed:
+
+```
+$ flox install python311Full
+✅ 'python311Full' installed to environment 'new-python-project'
+$ flox list
+python311Full: python311Full (python3-3.11.9)
+```
+
+When `flox upgrade` is run in this environment, the version of Python will be upgraded to the latest version available in the catalog. If you wish to pin your Flox environment to a specific version of Python, you can specify it in the `flox install` command:
```

Comment:
Note that it will only upgrade you to the latest Python 3.11, not the latest Python version. That's different and likely surprising so it's worth calling out. We had a similar discussion about Node.